content/docs/guide-development.md

```diff
@@ -258,6 +258,13 @@ os.environ['ODA_SECRET_STORAGE'] = '/secrets_custom'
 
 It is a good practice to test the developed notebook. This allows to make sure that the code remains valid in the future.
 A test is implemented as another notebook, except that name of the notebook starts with "test_". The notebook should call other notebooks and check that the output matches expectations. See an example of such a test [here](https://gitlab.renkulab.io/astronomy/mmoda/mmoda-nb2workflow-example/-/blob/master/notebooks/test_lightcurve.ipynb). 
+To run tests, it is necessary to add some dependencies to the basic environment: in `environment.yml`, add libmagic see [this example](https://gitlab.renkulab.io/astronomy/mmoda/jemx-expert/-/blob/main/environment.yml?ref_type=heads); in `requirements.txt` add the `dispatcher-plugin-nb2workflow` from the git repository, see the last line of [this example](https://gitlab.renkulab.io/astronomy/mmoda/jemx-expert/-/blob/main/requirements.txt?ref_type=heads). 
```

Comment:
Why `libmagic` here? Isn't it specific to `jemx-expert`? 

Concerning `dispatcher-plugin-nb2workflow`, it helps to look on how dispatcher "understands" the workflow and possible "catch" improper annotations. But it's not needed to run a test notebook. 
The downside of having it in the requirements is that it will be installed in the container, blowing its size and complicating dependencies.

Reply:
Hi Denys, thanks for this comment. We had already this discussion, but it is important to review our understanding. 
1. `libmagic` is required by nb2workflow to run the notebook possibly through the plugin (see https://github.com/oda-hub/nb2workflow/issues/199#issuecomment-2551817997  )
2. without `dispatcher-plugin-nb2workflow` nb2workflow run did not complete (see https://github.com/oda-hub/nb2workflow/issues/199#issuecomment-2328187720 )
I would be happy if this could simplify, but this is what we concluded trying to run a test on my notebooks.

Reply:
1. OK, this dependency problems need to be resolved in the package https://github.com/oda-hub/dispatcher-plugin-nb2workflow/issues/124
Having some dependencies pinned in the repo is sometimes necessary, but I wouldn't recommend putting this into docs. In this case, it's a kind of hotfix.

Reply:
2. This is just a warning, https://github.com/oda-hub/nb2workflow/blob/09c10f47fddd2a8327e89ea60bf5821c97268277/nb2workflow/nbadapter.py#L1145 
then all is skipped, unless `optional_dispather=False` [in nbrun](https://github.com/oda-hub/nb2workflow/blob/09c10f47fddd2a8327e89ea60bf5821c97268277/nb2workflow/nbadapter.py#L1193)
(controlled with `--mmoda_validation` [flag](https://github.com/oda-hub/nb2workflow/blob/09c10f47fddd2a8327e89ea60bf5821c97268277/nb2workflow/nbadapter.py#L1327) if run from command line)

Reply:
I now better understand this libmagic issue. `libmagic` itself is not a python package. Therefore, it's not possible to ensure its presence in the container via python package dependencies, only with conda (or modifying Dockerfile and installing with system package manager). Then the advise to install it makes sense. But it must be formulated the way it's obvious that it's only needed if there is a specific problem with dispatcher plugin.

And I still think dispatcher plugin is not strictly required there